models/GoogleNet.py

```diff
@@ -0,0 +1,115 @@
+    
+    x2 = layers.Conv2D(filters = f2_1,kernel_size = (1,1),strides = (1,1),padding = 'same',activation = 'relu')(x)
+    x2 = layers.Conv2D(filters = f2_2,kernel_size = (3,3),strides = (1,1),padding = 'same',activation = 'relu')(x2)
+    
+    x3 = layers.Conv2D(filters = f3_1,kernel_size = (1,1),strides = (1,1),padding = 'same',activation = 'relu')(x)
+    x3 = layers.Conv2D(filters = f3_2,kernel_size = (5,5),strides = (1,1),padding = 'same',activation = 'relu')(x3)
+    
+    x4 = layers.MaxPooling2D((3,3),strides = (1,1),padding = 'same')(x)
+    x4 = layers.Conv2D(filters = f4,kernel_size = (4,4),strides = (1,1),padding = 'same',activation = 'relu')(x4)
+    
+    x = layers.concatenate([x1, x2, x3, x4], axis=-1)
+    
+    return x
+
+
+inputs = keras.Input(shape = (32,32,3))
+x = layers.Conv2D(filters = 64,kernel_size = (7,7),strides = (2,2),padding = 'valid',activation = 'relu')(inputs)
+x = layers.MaxPooling2D((3,3),strides = 2,padding = 'same')(x)
+x = layers.Conv2D(filters = 64,kernel_size = (1,1),strides = (2,2),padding = 'same',activation = 'relu')(x)#local response normalisation
+
+x = layers.Conv2D(filters = 192,kernel_size = (3,2),strides = (1,1),padding = 'same',activation = 'relu')(x)
+x = layers.MaxPooling2D((3,3),strides = 2,padding = 'same')(x)
+
+x = inception_module(x,64,96,128,16,32,32)
+x = inception_module(x,128,128,192,32,96,64)
+x = layers.MaxPooling2D((3,3),strides = 2,padding = 'same')(x)
+x = inception_module(x,192,96,208,16,48,64)
+
+#first auxiliary classifier
+x1 = layers.AveragePooling2D((5,5),strides = 3,padding = 'same')(x)
+x1 = layers.Conv2D(filters = 128,kernel_size = (1,1),padding = 'same',activation = 'relu')(x1)
+x1 = layers.Flatten()(x1)
+x1 = layers.Dense(1024,activation = 'relu')(x1)
+x1 = layers.Dropout(0.7)(x1)
+x1 = layers.Dense(10,activation = 'softmax',name = 'auxiliary_1')(x1)
+
+x = inception_module(x,160,112,224,24,64,64)
+x = inception_module(x,128,128,256,24,64,64)
+x = inception_module(x,112,144,288,32,64,64)
+
+#second auxiliary classifier
+x2 = layers.AveragePooling2D((5,5),strides = 3,padding = 'same')(x)
+x2 = layers.Conv2D(filters = 128,kernel_size = (1,1),padding = 'same',activation = 'relu')(x2)
+x2 = layers.Flatten()(x2)
+x2 = layers.Dense(1024,activation = 'relu')(x2)
+x2 = layers.Dropout(0.7)(x2)
+x2 = layers.Dense(10,activation = 'softmax',name = 'auxiliary_2')(x2)
+
+
+x = inception_module(x,256,160,320,32,128,128)
+x = layers.MaxPooling2D((3,3),strides = (2,2),padding = 'same')(x)
+x = inception_module(x,256,160,320,32,128,128)
+x = inception_module(x,384,192,384,48,128,128)
+
+x = layers.GlobalAveragePooling2D()(x)
+x = layers.Dropout(0.4)(x)
+outputs = layers.Dense(10,activation = 'softmax',name = 'final_layer')(x)
+
+
+model = keras.Model(inputs = inputs,outputs = [outputs,x1,x2])
+model.summary()
+
+
+model.compile(optimizer = keras.optimizers.SGD(learning_rate = 0.1,momentum = 0.9,),
```

Comment:
Generally LR needs a very low value like `1e-3` or `1e-4`. But tuning it a bit is better.

Reply:
Use  [ml_collections](https://github.com/google/ml_collections) to maintain a config objects for all hyperparameters at the start of the file.

Reply:
Make sure to include this in all files.